pkg/scheduler/core/equivalence_cache.go

```diff
@@ -42,10 +38,10 @@ type EquivalenceCache struct {
 	algorithmCache map[string]AlgorithmCache
 }
 
-// The AlgorithmCache stores PredicateMap with predicate name as key
+// The AlgorithmCache stores PredicateMap with predicate name as key, PredicateMap as value.
 type AlgorithmCache struct {
-	// Only consider predicates for now
-	predicatesCache *lru.Cache
+	// Only consider predicates for now.
+	predicatesCache map[string]PredicateMap
```

Comment:
Suggest `type AlgorithmCache = map[string]PredicateMap` instead.

Reply:
Actually, having them all be maps lets us take advantage of nil map semantics in some fun ways.
```
result, ok := cache[nodeName][predicateName][equivHash]  // works with nil maps
delete(cache[nodeName], predicateName)  // works with nil maps
```

I'm new to golang and this is fun.

Reply:
Emm, I am not sure of this before actually. Just updated the code :-)